src/tables.jl

```diff
@@ -0,0 +1,73 @@
+"""
+Constructs `Table` out of `FeatureLayer`, where every row is a `Feature` consisting of Geometry and attributes.
+```
+ArchGDAL.Table(T::Union{IFeatureLayer, FeatureLayer})
+```
+"""
+struct Table{T<:Union{IFeatureLayer, FeatureLayer}}
+    layer::T
+end
+
+getlayer(t::Table) = Base.getfield(t, :layer)
+
+function Tables.schema(layer::AbstractFeatureLayer)
+    field_names, geom_names, featuredefn, fielddefns = schema_names(layer)
+    ngeom = ArchGDAL.ngeom(featuredefn)
+    geomdefns = (ArchGDAL.getgeomdefn(featuredefn, i) for i in 0:ngeom-1)
+    field_types = (_FIELDTYPE[gettype(fielddefn)] for fielddefn in fielddefns)
+    geom_types = (IGeometry for i in 1:ngeom)
+    Tables.Schema((field_names..., geom_names...), (field_types..., geom_types...))
+end
+
+Tables.istable(::Type{<:Table}) = true
+Tables.rowaccess(::Type{<:Table}) = true
+Tables.rows(t::Table) = t
+
+function Base.iterate(t::Table, st = 0)
+    layer = getlayer(t)
+    st >= nfeature(layer) && return nothing
+    if iszero(st)
+        resetreading!(layer)
+    end
+    return nextnamedtuple(layer), st + 1
+end
+
+function Base.getindex(t::Table, idx::Int)
```

Comment:
Should it be `idx::Integer`?

See the [AppVeyor failure](https://ci.appveyor.com/project/NgYeeSian/archgdal-jl/builds/35181016/job/57p2lmp40olma6c9):
```
Misc. methods: Error During Test at C:\projects\archgdal-jl\test\test_tables.jl:58
  Test threw exception
  Expression: typeof([getindex(gt1, i) for i = 1:size(gt1)]) == typeof([(iterate(gt1, i))[1] for i = 0:size(gt1) - 1])
  MethodError: no method matching getindex(::ArchGDAL.Table{ArchGDAL.IFeatureLayer}, ::Int64)
  Closest candidates are:
    getindex(::ArchGDAL.Table, !Matched::Int32) at C:\projects\archgdal-jl\src\tables.jl:36
  Stacktrace:
   [1] (::var"#11#19"{ArchGDAL.Table{ArchGDAL.IFeatureLayer}})(::Int64) at .\none:0
   [2] iterate at .\generator.jl:47 [inlined]
   [3] collect(::Base.Generator{UnitRange{Int64},var"#11#19"{ArchGDAL.Table{ArchGDAL.IFeatureLayer}}}) at .\array.jl:622
   [4] top-level scope at C:\projects\archgdal-jl\test\test_tables.jl:58
   [5] top-level scope at D:\buildbot\worker\package_win32\build\usr\share\julia\stdlib\v1.3\Test\src\Test.jl:1107
   [6] top-level scope at C:\projects\archgdal-jl\test\test_tables.jl:39
   [7] top-level scope at D:\buildbot\worker\package_win32\build\usr\share\julia\stdlib\v1.3\Test\src\Test.jl:1107
   [8] top-level scope at C:\projects\archgdal-jl\test\test_tables.jl:6
```

Reply:
Oh sorry I ignored it thinking that the failures were x86 expecting it to be an internal issue.